CMakeLists.txt

```diff
@@ -65,33 +93,39 @@ function(trusted_firmware_build)
     tfm
     SOURCE_DIR ${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m
     BINARY_DIR ${TFM_BINARY_DIR}
-    CMAKE_ARGS -DPROJ_CONFIG=${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/configs/${TFM_CFGFILE}.cmake
-               -DTARGET_PLATFORM=${TFM_BOARD}
+    CMAKE_ARGS -DCMAKE_TOOLCHAIN_FILE=${ZEPHYR_TFM_MODULE_DIR}/${CMAKE_TOOLCHAIN_FILE}
+               -DTFM_PLATFORM=${TFM_BOARD}
+               ${TFM_CMAKE_BUILD_TYPE_ARG}
                ${TFM_BL2_ARG}
-               -DCOMPILER=${TFM_TOOLCHAIN}
-               -DGNUARM_PREFIX=${TFM_TOOLCHAIN_PREFIX}
-               -DGNUARM_PATH=${TFM_TOOLCHAIN_PATH}
+               ${TFM_IPC_ARG}
+               ${TFM_ISOLATION_LEVEL_ARG}
+               ${TFM_REGRESSION_ARG}
+               -DTFM_TEST_REPO_PATH=${ZEPHYR_TFM_MODULE_DIR}/tf-m-tests
+               -DTFM_TOOLCHAIN_PATH=${TFM_TOOLCHAIN_PATH}
+               -DTFM_TOOLCHAIN_PREFIX=${TFM_TOOLCHAIN_PREFIX}
     BUILD_ALWAYS True
     USES_TERMINAL_BUILD True
-    BUILD_BYPRODUCTS ${VENEERS_FILE}
+    BUILD_BYPRODUCTS ${VENEERS_FILE} ${PSA_API_NS_PATH}
   )
 
-  set(TFM_INTERFACE_SOURCE ${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/interface/src)
-
   # IPC mode source dependencies
-  if(TFM_IPC)
-    add_library(tfm_ipc_psa_api
-                ${TFM_INTERFACE_SOURCE}/tfm_psa_ns_api.c
-                ${TFM_INTERFACE_SOURCE}/tfm_crypto_ipc_api.c
-                ${TFM_INTERFACE_SOURCE}/tfm_initial_attestation_ipc_api.c
-                ${TFM_INTERFACE_SOURCE}/tfm_its_ipc_api.c
-                ${TFM_INTERFACE_SOURCE}/tfm_platform_ipc_api.c
-                ${TFM_INTERFACE_SOURCE}/tfm_ps_ipc_api.c
-                ${TFM_INTERFACE_SOURCE}/tfm_ns_interface.c
-    )
-    target_include_directories(tfm_ipc_psa_api PUBLIC ${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/interface/include)
-    target_compile_definitions(tfm_ipc_psa_api PUBLIC TFM_PSA_API TFM_PARTITION_TEST_CORE_IPC)
-    target_link_libraries(tfm_ipc_psa_api PRIVATE zephyr_interface)
-  endif()
+  add_library(tfm_ipc_psa_api STATIC IMPORTED)
```

Comment:
We should probably change the name since both ipc and psa are configurable.

Maybe just "tfm_api" or "tfm_interface"?

Reply:
tfm_api is good.
Will apply.

Reply:
Done